apps/user/services/authentication.go

```diff
@@ -90,6 +95,103 @@ func (service AuthenticationService) BasicLogin(c *gin.Context) (
 	return userResponse, http.StatusAccepted, nil
 }
 
+// Identify ...
+func (service AuthenticationService) Identify(c *gin.Context) (
+	status int,
+	err error) {
+
+	var identifyData userSerializers.Identify
+	err = c.BindJSON(&identifyData)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+	gormDB := service.DB
+	var userData userModels.User
+	err = gormDB.Model(&userModels.User{}).Where("email = ?", identifyData.Email).Scan(&userData).Error
+	if err != nil {
+		if gorm.IsRecordNotFoundError(err) {
+			return http.StatusBadRequest, fmt.Errorf("We couldn't find a iReflect account associated with %s", identifyData.Email)
+		}
+		return http.StatusInternalServerError, err
+	}
+	// when we don't need OTP to the email.
+	if !identifyData.SendOTP {
+		return http.StatusOK, nil
+	}
+	var otp userModels.OTP
+	err = gormDB.Model(&userModels.OTP{}).Where("user_id = ?", userData.ID).Scan(&otp).Error
+	if err != nil && !gorm.IsRecordNotFoundError(err) {
+		return http.StatusInternalServerError, err
+	}
+	// if OTP exists then check its validity.
+	if !gorm.IsRecordNotFoundError(err) {
```

Comment:
Doesn't the above condition `if err != nil && !gorm.IsRecordNotFoundError(err)` confirm that an OTP exists for the user?

Reply:
yes.